### PR TITLE
Minor: Add comment on input_schema from AggregateExec

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

None

## Rationale for this change

As we integrate `AggregateExec`, one thing confusing me is why `AggregateExec` keeps a separate `input_schema` which is not actually schema of input operator, but is always the schema of input of aggregation. In other words, partial and final aggregation operators have the same input schema parameter. It doesn't look consistent on `input_schema` and `input` functions on the operator and other operators. And another look at this `input_schema` parameter, it doesn't look like there is any meaningful usage of it. It causes a bit strange when we integrate Spark with DataFusion's aggregation operator.

I've tried to remove it first in this patch. It is confirmed that `input_schema` is not used during query execution as all unit tests and end-to-end tests passed without it.

However, `verify benchmark results` pipeline has some failures. After looking into it, I found that the parameter is used when generating physical plan from protobuf nodes. When doing that, it needs input schema of partial aggregation to initiate aggregate expressions.

I changed this patch to add a few comment on it to help me and others to understand it in the future.

### CODE REVIEW

**Comment by viirya** [comment]
Interesting. All tests are passed but `verify benchmark results` has failure.

**Comment by viirya** [comment]
Ah, I see. It is used when generating physical plan from proto nodes. When doing that, it needs input schema of partial aggregation to initiate aggregate expressions.

**Comment by viirya** [comment]
For the above reason, I'm not sure if it's possible to remove the parameter now. Let me think about it more.

**Comment by viirya** [comment]
cc @alamb

File: datafusion/physical-plan/src/aggregates/mod.rs [hunk 1]
```diff
@@ -285,7 +285,9 @@ pub struct AggregateExec {
     schema: SchemaRef,
     /// Input schema before any aggregation is applied. For partial aggregate this will be the
     /// same as input.schema() but for the final aggregate it will be the same as the input
-    /// to the partial aggregate
+    /// to the partial aggregate, i.e., partial and final aggregates have same `input_schema`.
```
Comment (alamb) [inline datafusion/physical-plan/src/aggregates/mod.rs:288]: 🤔  I went to double check this (being inspired by the thoroughness of your own reviews @viirya ) and I was able to remove the entire field: https://github.com/apache/arrow-datafusion/pull/7741

**Review by alamb — APPROVED** [review]
I think this change is fine, but I think https://github.com/apache/arrow-datafusion/pull/7741 would be even better

**Comment by viirya** [comment]
I am far from a laptop but I tried to remove it at the first and the schema
seems necessary when deserializing from protobuf to physical aggregate plan
as aggregate expression initialization needs it.

**Comment by viirya** [comment]
I left a comment above to explain it and where/why it is failed in the CI
pipeline.

**Comment by viirya** [comment]
@alamb You can see my previous commits which removed the `input_schema` field. In CI, all unit tests and end-to-end tests can pass but `verify benchmark results` gets failures when deserializing protobuf to physical aggregate. 

It is because during initializing aggregate expressions for final aggregate, their constructors all needs their input data type from input schema of partial aggregate (i.e., input schema before aggregate). That is why `input_schema` is there.

**Comment by alamb** [comment]
> @alamb You can see my previous commits which removed the `input_schema` field. In CI, all unit tests and end-to-end tests can pass but `verify benchmark results` gets failures when deserializing protobuf to physical aggregate.
> 
> It is because during initializing aggregate expressions for final aggregate, their constructors all needs their input data type from input schema of partial aggregate (i.e., input schema before aggregate). That is why `input_schema` is there.

Sorry @viirya  -- I should have looked at your comments more closely.

**Comment by viirya** [comment]
No worries @alamb . Thanks for approval. 😄 

I guess it is still possible to remove it, if we embed the data type info in to individual aggregate expression protobuf node. But this will involve a lot of changes just for removing the `input_schema`. 😄  So I stopped there and turned to add a few comment to explain why the parameter is there.

**Comment by alamb** [comment]
> I guess it is still possible to remove it, if we embed the data type info in to individual aggregate expression protobuf node. But this will involve a lot of changes just for removing the input_schema. 😄 So I stopped there and turned to add a few comment to explain why the parameter is there.

Makes sense. I was thinking we could effectively recover the intermediate schema somehow (though I not quite sure how)

**Comment by viirya** [comment]
Thanks @alamb. I merged this first. We may investigate if it is possible to remove it completely later.

**Pull request merged**